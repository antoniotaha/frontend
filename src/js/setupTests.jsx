import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import httpAdapter from 'axios/lib/adapters/http'
import axios from 'axios'

export function shallowRenderAndMatch(componentJsx) {
	const wrapper = shallow(componentJsx)
	expect(wrapper).toMatchSnapshot()
	return wrapper
}

Enzyme.configure({ adapter: new Adapter() })
axios.defaults.adapter = httpAdapter 
