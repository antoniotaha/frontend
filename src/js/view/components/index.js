
export * from './elements/Button'
export * from './elements/Card'
export * from './elements/Image'
export * from './elements/Input'
export * from './elements/Label'
export * from './elements/Pagination'
export * from './elements/TextArea'
export * from './elements/TextInput'


export * from './form/Control'
export * from './form/Field'

export * from './grid/Column'
export * from './grid/Columns'

export * from './Header'
export * from './Conteudo'

export * from './usuario/CardRepositorio'
export * from './usuario/PageNotFound'


