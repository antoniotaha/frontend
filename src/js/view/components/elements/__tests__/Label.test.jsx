import * as React from 'react'
import * as C from 'view/components'
import { shallowRenderAndMatch } from 'setupTests'

describe('Label', () => {
	it('render', () => {
		shallowRenderAndMatch(<C.Label />)
	})

	it('render custom classes', () => {
		shallowRenderAndMatch(<C.Label className='test' />)
	})

	it('render label com texto ', () => {
		shallowRenderAndMatch(<C.Label>Teste</C.Label>)
	})
})
