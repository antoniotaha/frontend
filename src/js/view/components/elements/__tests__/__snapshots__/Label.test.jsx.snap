// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "label",
    },
    "ref": null,
    "rendered": null,
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "label",
      },
      "ref": null,
      "rendered": null,
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Label render custom classes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    className="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "label test",
    },
    "ref": null,
    "rendered": null,
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "label test",
      },
      "ref": null,
      "rendered": null,
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Label render label com texto  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label>
    Teste
</Label>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "Teste",
      "className": "label",
    },
    "ref": null,
    "rendered": "Teste",
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Teste",
        "className": "label",
      },
      "ref": null,
      "rendered": "Teste",
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
